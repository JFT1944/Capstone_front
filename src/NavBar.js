import React from "react";
import { NavLink, useNavigate } from "react-router-dom";


function NavBar(props){
let {user, token, setUser, setToken} = props
let navigate = useNavigate()

function LogOutSequence(e){
e.preventDefault()

setUser(undefined)
setToken(undefined)
localStorage.clear()
navigate('/')

}

function handleMobileNav(e){
    e.preventDefault()
    navigate('/mobile-nav')
}


    if(user && token){
        return(
            <>
            <div className="logo">
                <span className="logo1">Dispensa<span className="logo2">.io</span></span>
            </div>
            <div className="NavBar_holder">
                <NavLink className='navs' exact={true} to='/part-form' data-testid='single'>Add Single Ingredient</NavLink>
                <NavLink className='navs' exact={true} to='/full-form'>Add Recipe Ingredients</NavLink>
                <NavLink className='navs' exact={true} to='/recipes'>Recipes</NavLink>
                <NavLink className='navs' exact={true} to='/pantry'>Pantry</NavLink>
                <NavLink className='navs' exact={true} to='/shopping-list'>Shopping List</NavLink>
                <NavLink className='navs' exact={true} to='/' onClick={(e) => LogOutSequence(e)}>Log Out</NavLink>
                
            </div>
            <div className="mobile_button" onClick={(e) => handleMobileNav(e) }>
                <img alt="Pantry Shelving Icon"  src="https://cdn3.iconfinder.com/data/icons/food-technology-2/64/Shelves_Food-shelving-supermarket-drink-512.png"/>
            </div>
            </>
        )
    }else{


    return(
        <>
        <div className="logo">
                <span className="logo1">Dispensa<span className="logo2">.io</span></span>
            </div>
        <div className="NavBar_holder">
            <NavLink className='navs' exact={true} to='/' >Home</NavLink>
            <NavLink className='navs' exact={true} to='/login' >Login</NavLink>
            <NavLink className='navs' exact={true} to='/signup' >Sign Up</NavLink>
        </div>
        </>
    )}
}


export default NavBar