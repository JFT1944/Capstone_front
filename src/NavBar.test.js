
import React from "react";
import { render, fireEvent, getByTestId } from "@testing-library/react";
import NavBar from "./NavBar";
import { BrowserRouter, MemoryRouter } from "react-router-dom";


// write a smoke test for the home component

it('should appear in a smoketest', () => {
    <BrowserRouter>
    render(<NavBar user={'user'} setUser={'setUser'} token={'token'} setToken={'setToken'}/>)
    </BrowserRouter>
})

// snapshot test
it('should match snapshot', () => {
    const {asFragment} = render(<MemoryRouter><NavBar user={'user'} token={'token'}/></MemoryRouter>)
    expect(asFragment()).toMatchSnapshot()
})