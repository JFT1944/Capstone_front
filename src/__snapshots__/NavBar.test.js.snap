// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="logo"
  >
    <span
      class="logo1"
    >
      Dispensa
      <span
        class="logo2"
      >
        .io
      </span>
    </span>
  </div>
  <div
    class="NavBar_holder"
  >
    <a
      class="navs"
      data-testid="single"
      href="/part-form"
    >
      Add Single Ingredient
    </a>
    <a
      class="navs"
      href="/full-form"
    >
      Add Recipe Ingredients
    </a>
    <a
      class="navs"
      href="/recipes"
    >
      Recipes
    </a>
    <a
      class="navs"
      href="/pantry"
    >
      Pantry
    </a>
    <a
      class="navs"
      href="/shopping-list"
    >
      Shopping List
    </a>
    <a
      aria-current="page"
      class="navs active"
      href="/"
    >
      Log Out
    </a>
  </div>
  <div
    class="mobile_button"
  >
    <img
      alt="Pantry Shelving Icon"
      src="https://cdn3.iconfinder.com/data/icons/food-technology-2/64/Shelves_Food-shelving-supermarket-drink-512.png"
    />
  </div>
</DocumentFragment>
`;
